js/views/conversation_view.js

```diff
@@ -2186,12 +2194,183 @@
       }
     },
 
+    handleDeleteOrBackspace(event, isDelete) {
+      const $input = this.$messageField[0];
+      const text = this.$messageField.val();
+
+      // Only handle the case when nothing is selected
+      if ($input.selectionDirection !== 'none') {
+        // Note: if this ends up deleting a handle, we should
+        // (ideally) check if we need to update the mapping in
+        // `this.memberView`, but that's not vital as we already
+        // reset it on every 'send'
+        return;
+      }
+
+      const mentions = this.memberView.pendingMentions();
+
+      const _ = window.Lodash; // no underscore.js please
+      const predicate = isDelete ? _.startsWith : _.endsWith;
+
+      const pos = $input.selectionStart;
+      const part = isDelete ? text.substr(pos) : text.substr(0, pos);
```

Comment:
Do we need to handle the case when users select multiple character and presses delete or backspace?

Reply:
This makes sure we fall back to the default behaviour if multiple characters are selected: `($input.selectionDirection !== 'none') return`. That's because if there are multiple selected characters, we don't delete the previous/following character, but rather delete the selection, so we don't need to do anything special. Note that if we assume the user does not use the mouse cursor to select part of the message, there is no way to select a mention partly.

Correct handling of the mouse selection doesn't seem very important at the moment, so I omitted that in the interest of time (but we could revisit this later).

Reply:
Sounds good to me ;)